src/widgets/data_generator_widget.py

```diff
@@ -0,0 +1,116 @@
+from functools import partial
+from typing import Dict, Tuple, Type, Callable
+
+from PyQt5.QtWidgets import QWidget, QHBoxLayout, QVBoxLayout, QPushButton, QGroupBox, QComboBox, QLabel, QFormLayout, \
+    QTableView, QSizePolicy
+from PyQt5.QtCore import QRect
+
+from data_generators import clustering_blobs_generator, DataGeneratorFunction
+from engines import ImportDataEngine
+from widgets import QtTable
+from widgets.options_widgets import ClusteringBlobsDataOptions, Options
+
+
+class DataGeneratorWidget(QWidget):
+    def __init__(self, engine: ImportDataEngine, callback: Callable):
+        super().__init__()
+        self.engine = engine
+        self.callback = callback
+
+        self.generated_data = None
+        self.setWindowTitle("Data Generator")
+        self.setGeometry(QRect(400, 400, 800, 400))
+
+        self.setObjectName("data_generator_widget")
+        with open('../static/css/styles.css') as stylesheet:
+            self.setStyleSheet(stylesheet.read())
+
+        self.dataset_types_config: Dict[str, Tuple[DataGeneratorFunction, Type[Options]]] = {
+            "(Clustering) Normal distribution blobs": (clustering_blobs_generator, ClusteringBlobsDataOptions),
+        }
+
+        self.layout = QHBoxLayout()
+        self._render_algorithm_selection()
+        self._render_options()
+
+        self.generate_button = QPushButton(self)
+        self.generate_button.setText("Generate")
+        self.generate_button.clicked.connect(partial(self.click_listener, 'generate'))
+
+        self.save_button = QPushButton(self)
+        self.save_button.setText("Save")
```

Comment:
Maybe it should be "Load", because app doesn't save this data to file or database.